business_rules/utils.py

```diff
@@ -120,6 +127,29 @@ def check_params_valid_for_method(method, given_params, method_type_name):
             ', '.join(invalid_params), method_type_name, method.__name__))
 
 
+def check_for_default_value_for_missing_params(missing_params, method_params):
+    """
+    :param missing_params: Params missing from Rule
+    :param method_params: Params defined on method, which could have default value for missing param
+    [{
+     'label': 'release voucher from template',
+     'name': 'voucher_template_id',
+     'fieldType': 'numeric',
+     'defaultValue': 123
+    },
+    ...
+    ]
+    :return Params that are missing from rule but have default params: {'voucher_template_id'}
+    """
+    params_with_default_value = set()
+    if method_params:
+        for param in method_params:
+            if param['name'] in missing_params and param.get('defaultValue') is not None:
```

Comment:
I'm worried what would happen if a action definition doesn't have a name defined... You should try to validate this when defining a rule action 🤔

Reply:
`name` key is added in `@rule_action` decorator, it won't be added only if rule parameters' dict is empty, in this case `method_params` will become an empty list. Thought that if otherwise `name` is not present it should blow up here indicating something is wrong?